doc/scenario.md

```diff
@@ -24,43 +24,146 @@ In `cartridge_scenario` you should specify names of steps. The default scenario
+### Editing topology without connecting to membership
+
+For example, you want to edit topology, but you don't want to wait for a long `connect_to_membership` step
+(in fact this step is only needed to find some instance that is already in cluster).
+
+As we can see in `edit_topology` [step API](#edit_topology), it requires `control_instqnce` fact. Since we definitely
+know a name of instance that is already in cluster, we can specify it explicitly and remove `connect_to_membership` step
+from scenario:
+```yaml
+# edit_topology_playbook.yml
+
+- name: Edit topology of my cluster
+  hosts: all
+  tasks:
+    - name: Edit topology by core 1
+      import_role:
+        name: tarantool.cartridge
+      vars:
+        cartridge_scenario:
+          - edit_topology
+        control_instance:
+          name: core_1
+          console_sock: '/var/run/tarantool/core_1.control'
+```
+
+### Replace copying package to remote host
+
+By default, package specified by `cartridge_package_path` is simple copied to remote host, it's happened
+on `deliver_package` step. Sometimes we need to download or copy package from some custom place. Let's do it!
+
+Generally, we just need to override default `deliver_package` step.
+
+Let's take a look on it's [API](#deliver_package). It requires two variables:
+- `cartridge_package_path` that we simply specify in vars;
+- `single_instances_for_each_machine` that allows us to run this task once per each machine.
+
+As a result of this module we should set `delivered_package_path` variable
+(a path of package on remote machine).
+
+Now, choose a directory where our custom steps are placed, for example `./custom_steps`.
+
+Then create `./custom_steps/deliver_package.yml` and describe a way to deliver package:
 ```yaml
-cartridge_scenario:
-  - edit_topology
-  - cleanup_expelled
+# ./custom_steps/deliver_package.yml
+
+- when:
+    - cartridge_package_path is not none
+    - inventory_hostname in single_instances_for_each_machine
+  tags: cartridge-instances
+  block:
+    - name: 'Download package'
+      get_url:
+        url: 'https://some-server/myapp-1.0.0-0.rpm'
+        dest: '/tmp'
+      register: downloaded_package
+
+    - name: 'Set delivered package path'
+      set_fact:
+        delivered_package_path: '{{ downloaded_package.dest }}'
 ```
 
-Also, you can specify control instance in config for `edit_topology` step:
+Import role and say where to find our custom steps:
 ```yaml
-control_instance:
-  name: core_1
-  console_sock: '/var/run/tarantool/core_1.control'
+# deploy_application.yml
+
+- name: My playbook
+  hosts: all
+  tasks:
+    - name: Deploy application
+      import_role:
+        name: tarantool.cartridge
+      vars:
+        cartridge_custom_steps_dir: "./custom_steps"
 ```
 
-If you want to replace `deliver_package` step with a custom one, you should set a path to a directory
-with `deliver_package.yml` file to the variable `cartridge_custom_steps_dir`:
+### Replace instances restarting
```

Comment:
I think that this example can be replaced with two little examples:

### Adding custom step to scenario

You can create your own step and include it to scenario.
Let's create `./custom_steps/special_magic.yml`. 
After that, you can add `special_magic` step to scenario. 
Just pass path to a directory where your custom steps are placed:

```
- name: My playbook
  hosts: all
  tasks:
    - name: Deploy application
      import_role:
        name: tarantool.cartridge
      vars:
        cartridge_custom_steps_dir: "./custom_steps"
        cartridge_scenario:
          - restart_instance
          - special_magic
```

### Importing steps from different directories

If you want to import steps by full paths, `cartridge_custom_steps` parameter can be useful.
It allows to specify mapping between step name and path to file to import:

```yaml
cartridge_custom_steps:
  - name: 'common_special_task'
     file: './common/special_task.yml'
```